src/optimizer.cpp

```diff
@@ -41,24 +41,24 @@ void Optimizer::getParams()
   auto node = parent_.lock();
 
   auto getParam = utils::getParamGetter(node, name_);
+  auto getParentParam = utils::getParamGetter(node, "");
 
   getParam(model_dt_, "model_dt", 0.1);
   getParam(time_steps_, "time_steps", 15);
   getParam(batch_size_, "batch_size", 200);
   getParam(iteration_count_, "iteration_count", 2);
   getParam(temperature_, "temperature", 0.25);
 
-  getParam(constraints_.vx, "vx_max", 0.5);
-  getParam(constraints_.vy, "vy_max", 1.3);
-  getParam(constraints_.vw, "wz_max", 1.3);
-  getParam(vx_std_, "vx_std", 0.1);
-  getParam(vy_std_, "vy_std", 0.1);
-  getParam(wz_std_, "wz_std", 0.3);
-  getParam(control_sequence_shift_offset_, "control_sequence_shift_offset", 1);
+  getParam(default_constraints_.vx, "vx_max", 0.5);
+  getParam(default_constraints_.vy, "vy_max", 1.3);
+  getParam(default_constraints_.vw, "wz_max", 1.3);
+  constraints_ = default_constraints_;
+
+  getParam(sampling_std_.vx, "vx_std", 0.1);
+  getParam(sampling_std_.vy, "vy_std", 0.1);
+  getParam(sampling_std_.wz, "wz_std", 0.3);
+  getParentParam(controller_frequency_, "controller_frequency", 0.0);
```

Comment:
Please revert this to the 

```
  nav2_util::declare_parameter_if_not_declared(
    node, "controller_frequency", rclcpp::ParameterValue(0.0));
  node->get_parameter("controller_frequency", controller_frequency_);
```

block, I'm not sure why, but it was causing testing warnings about temporaries that I didn't fully understand but doing the usual parameter declaration worked well

Reply:
I found a way to get around it